### PR TITLE
Add and fix several Cisco and H3C devices

### PR DESCRIPTION
**Pull request by alextremblay**
As per #1171

### CODE REVIEW

File: device-types/Cisco/WS-C3650-48FQM-S.yaml
Comment (danner26) [inline device-types/Cisco/WS-C3650-48FQM-S.yaml:216]: It looks like these PSUs are modular. Please remove these definitions and create `module-bays` for these. The fans should also be created in a similar fashion

Then please add the applicable module-type files if they do not already exist

**Comment by danner26** [comment]
Looks like there are some issues with the new addition, so that is going to delay merging. If you want to break this out into 2 separate PRs, ping me when you remove the new file and I will merge the fix.. Then you can create a new PR for that new file

Or just leave it and lets get the fixes in place, then we can merge it all. Your call

Thank you for the fix btw!

**Comment by danner26** [comment]
Hello, unfortunately today we had to make a change to the schema in which weight is defined by. The upstream NetBox API requires weight to be passed to it in the following format:
```
weight: 12.01
weight_unit: lb
```

Due to this change, I have updated your branch against the current master. This might require you to update your PR, if you had weight currently defined, so that your PR is in-line with the new weight schema. If you have any issues, please let me know or refer to [this PR](https://github.com/netbox-community/devicetype-library/pull/1187) for more information.

File: device-types/Cisco/WS-C3650-48FQM-S.yaml
Comment (danner26) [reply device-types/Cisco/WS-C3650-48FQM-S.yaml:4]: We are standardizing the repository. All device slugs from hereon out will need to have their manufacturer name as a prefix. If you'd like to assist, all of the Cisco devices will need this done to them as well

File: device-types/Cisco/WS-C3650-48FQM-S.yaml
Comment (alextremblay) [reply device-types/Cisco/WS-C3650-48FQM-S.yaml:10]: done

File: device-types/Cisco/WS-C3650-48FQM-S.yaml
Comment (alextremblay) [reply device-types/Cisco/WS-C3650-48FQM-S.yaml]: done

File: device-types/Cisco/WS-C3650-48FQM-S.yaml
Comment (alextremblay) [reply device-types/Cisco/WS-C3650-48FQM-S.yaml:4]: ahhh, i see.

Ok, will do.

**Comment by danner26** [comment]
I am going to convert this to a draft.. When you are done working feel free to mark it ready for review/@ me and I will re-review! Thanks for all you are doing

**Comment by alextremblay** [comment]
Interesting, the build failure on CI did not occur when I ran pytest locally...

ahhhh, missing yaml document markers. ok, I'll fix

**Comment by alextremblay** [comment]
Hi @danner26 , I think this one's ready for primetime. Can you please review/approve?

**Comment by danner26** [comment]
> Hi @danner26 , I think this one's ready for primetime. Can you please review/approve?

It is a big one! I am moving through the changes slowly but surely

File: device-types/Cisco/WS-C3850-48F-L.yaml
Comment (danner26) [inline device-types/Cisco/WS-C3850-48F-L.yaml:211]: This appears to be a modular PSU. These should be put under module-bays and you should also create the fans as modules as well

File: device-types/Cisco/WS-C3850-48F-L.yaml
Comment (danner26) [inline device-types/Cisco/WS-C3850-48F-L.yaml:214]: This appears to be a modular PSU. These should be put under module-bays and you should also create the fans as modules as well

File: device-types/Cisco/WS-C3850-12XS-S.yaml
Comment (danner26) [inline device-types/Cisco/WS-C3850-12XS-S.yaml:44]: This appears to be a modular PSU. These should be put under module-bays and you should also create the fans as modules as well

It also appears there are 2 for this model

File: device-types/Cisco/WS-C3850-12XS-E.yaml
Comment (danner26) [inline device-types/Cisco/WS-C3850-12XS-E.yaml:44]: This appears to be a modular PSU. These should be put under module-bays and you should also create the fans as modules as well

It also appears there are 2 for this model

File: device-types/Cisco/WS-C3750X-48P-L.yaml
Comment (danner26) [inline device-types/Cisco/WS-C3750X-48P-L.yaml:215]: This appears to be a modular PSU. These should be put under module-bays and you should also create the fans as modules as well

File: device-types/Cisco/WS-C3750X-48P-L.yaml
Comment (danner26) [inline device-types/Cisco/WS-C3750X-48P-L.yaml:218]: This appears to be a modular PSU. These should be put under module-bays and you should also create the fans as modules as well

File: device-types/Cisco/WS-C3650-24PDM-L.yaml
Comment (danner26) [inline device-types/Cisco/WS-C3650-24PDM-L.yaml:122]: These also have a secondary dc backup slot if I am not mistaken, please add that as well

File: device-types/Cisco/WS-C3650-24PDM-L.yaml
Comment (danner26) [inline device-types/Cisco/WS-C3650-24PDM-L.yaml:7]: Please add `module-bays` with the 2 stacking ports

**Review by danner26 — CHANGES_REQUESTED** [review]
Please review the corrections/changes made as well as the comments regarding other required changes. If you have any questions let me know. In the meantime I will mark this as a draft PR again until you are ready for re-review. Once you are ready feel free to ping me and mark this as ready for review. Thanks!

File: module-types/Cisco/PWR-C2-250WAC.yaml
Comment (alextremblay) [reply module-types/Cisco/PWR-C2-250WAC.yaml]: I see you've already made this change. Thank you!

File: module-types/Cisco/PWR-C2-640WDC.yaml
Comment (alextremblay) [reply module-types/Cisco/PWR-C2-640WDC.yaml]: I see you've already made this change. Thank you!

File: device-types/Cisco/WS-C3850-48XS-E.yaml
Comment (alextremblay) [reply device-types/Cisco/WS-C3850-48XS-E.yaml:117]: I'm working on making the changes you requested, but the scope of this PR has been steadily creeping larger and larger.

My original goal with this PR was to add devices which were missing from the library, adapting from similar devices which were already in the library. I will make the power port change you requested, but changing making all the fans of all these models modular is too much for me at the moment. The existing C3850 models in the library have fans, instead of fan modules, and so the devices i've added also have fans instead of fan modules. I do not have time to fix these pre-existing discrepancies right now.

Sorry.

File: device-types/Cisco/WS-C3650-24PDM-L.yaml
Comment (alextremblay) [reply device-types/Cisco/WS-C3650-24PDM-L.yaml:7]: All C3650, C3750X, and C3850 models support stacking cables, according to the data sheets, and yet none of these pre-existing device type files make any mention of stacking modules. Such a change is beyond the scope of this PR / issue

I think a separate issue / PR should be opened to address that.

File: device-types/Cisco/WS-C3650-24PDM-L.yaml
Comment (alextremblay) [reply device-types/Cisco/WS-C3650-24PDM-L.yaml:122]: If that's true of this model, it's likely true of all the 'minis' across the C3650, C3750X, and C3850 product lines. We'll have to comb through all the data sheets and confirm. Such a change is beyond the scope of this PR / issue

I think a separate issue / PR should be opened to address that.

**Comment by alextremblay** [comment]
Hi @danner26 , This PR is now ready for review.

Please see my comments regarding scope creep. I hope you see things as I do and agree with my sentiment. as much as I want this PR to be as good and as accurate as it can be, I've already sunk considerable time and effort into this endeavour, far beyond what the original PR / issue called for, and I have no more time or energy to give.

I hope you will accept this PR as it is and open separate issues for the fan modules, stakcing modules, and dc-terminal power supplies as mentioned in the comments.

If further changes are needed in order for this PR to be merged, I'm afreaid you'll have to find someone else to make them :(

File: device-types/Cisco/WS-C3850-48XS-E.yaml
Comment (danner26) [reply device-types/Cisco/WS-C3850-48XS-E.yaml]: I have to admit I was surprised that you added the `cisco-` slug changes in this PR, I would have expected them to be made in another PR. That being said, I went through and reviewed all of the changes you had added.

Also, I was just mentioning that the fans which are modular should be added to the module-bays, not specifically that you need to make that change.. I am sorry if I wasnt clear about that.. It was more of a statement than a request. The PSU changes are the only important ones here

Comment (danner26) [reply device-types/Cisco/WS-C3850-48XS-E.yaml]: I am sorry for the confusion regarding that.

File: device-types/Cisco/WS-C3650-24PDM-L.yaml
Comment (danner26) [reply device-types/Cisco/WS-C3650-24PDM-L.yaml:122]: I absolutely agree, but any new files you are adding should include all of the applicable power-ports, interfaces, etc

File: device-types/Cisco/WS-C3650-24PDM-L.yaml
Comment (DanSheps) [reply device-types/Cisco/WS-C3650-24PDM-L.yaml:7]: Stack ports are fixed ports, they shouldn't be modular

Comment (danner26) [reply device-types/Cisco/WS-C3650-24PDM-L.yaml:7]: Dan and I had a conversation, I should actually change how I phrase this in the future. If the stacking ports are modular, then they should be in module-bays.. Otherwise they should be in interfaces

Thanks for the catch Dan

Comment (DanSheps) [reply device-types/Cisco/WS-C3650-24PDM-L.yaml:7]: It does look like the "L" versions are modular, but the 3750X, and straight 3850's are non-modular.  Not sure about the 3650's but I imagine they are non-modular as well.

File: device-types/Cisco/WS-C3650-24PDM-L.yaml
Comment (danner26) [reply device-types/Cisco/WS-C3650-24PDM-L.yaml:122]: Adding the RPS 2300 port

**Comment by danner26** [comment]
Hey @alextremblay all looks good to me now, if you'd like, please review the changes I have made and let me know if you approve

@DanSheps if you would like to look through as well feel free! Once Alex gives me the go ahead I will merge this

**Comment by alextremblay** [comment]
Awesome, that all looks great!

Thank you so much for your help on this PR

**Pull request merged**
